src/app/_shared-components/PostComments/Comments/Comments.tsx

```diff
@@ -89,20 +106,31 @@ function Comments({
 		if (!hash) return;
 
 		const commentId = hash.replace('#comment-', '');
-		const comment = regularComments.find((c) => c.id === commentId);
+		const comment = [...verifiedComments, ...unverifiedComments, ...spamComments].find((c) => c.id === commentId);
 
-		if (comment && !showMore && regularComments.indexOf(comment) >= 2) {
+		if (!comment) return;
+
+		const verifiedCommentIndex = verifiedComments.findIndex((c) => c.id === comment.id);
+		if (verifiedCommentIndex !== -1 && !showMore && verifiedCommentIndex >= 2) {
 			handleShowMore();
 		}
 
+		if (unverifiedComments.some((c) => c.id === comment.id) && !showUnverified) {
+			setShowUnverified(true);
+		}
+
+		if (spamComments.some((c) => c.id === comment.id) && !showSpam) {
+			setShowSpam(true);
+		}
+
 		// Use requestAnimationFrame to ensure the DOM is ready
 		requestAnimationFrame(() => {
 			const element = document.getElementById(hash.substring(1));
 			if (element) {
 				element.scrollIntoView({ behavior: 'smooth', block: 'center' });
 			}
 		});
-	}, [regularComments, showMore]);
+	}, [verifiedComments, unverifiedComments, spamComments, showMore, showUnverified, showSpam]);
```

Comment:
this will be called again and again due to the states showUnverified and showSpam.